### ISSUE TITLE
Client Metrics does not use template URI variables

### ISSUE THREAD

**Issue by mmaggioni-wise** [quarkusio/quarkus#15231]
**Description**

Client Metrics does not use template URI variable for the called api.

instead of having one metric: 
`http_client_requests_seconds_count{.... status="200",uri="/api/v1/surveys/{surveyId}",} 5.0`

there are multiple metrics:
`http_client_requests_seconds_count{.... status="200",uri="/api/v1/surveys/387",} 3.0`
`http_client_requests_seconds_count{.... status="200",uri="/api/v1/surveys/388",} 2.0`

I don't think it's the correct behavior, it should probably works like the server metrics. 

I did a very little debugging (i m not very familiar with the code), and the problem seems in the HttpRequestMetric, the `routingContext` variable is null, so he ll just return the complete path.

It may be a problem with my dependencies?

Thanks

**Comment by jmartisk** [comment]
@ebullient

**Comment by ebullient** [comment]
I have an existing PR in flight for improved URI templating, #15047 
Can you confirm the version of Quarkus and extensions you're using?

To handle this in the meanwhile, you can set a matchPattern: 
`quarkus.micrometer.binder.http-client.match-pattern=/api/v1/surveys/[0-9]+=/api/v1/surveys/{surveyId}

**Comment by mmaggioni-wise** [comment]
Thank you!

Quarkus version is the 1.12.0.Final, the extensions are this ones:

`cdi, jaeger, micrometer, mutiny, rest-client, resteasy, resteasy-jsonb, resteasy-multipart, security, security-properties-file, smallrye-context-propagation, smallrye-fault-tolerance, smallrye-health, smallrye-jwt, smallrye-openapi, smallrye-opentracing, swagger-ui, vertx, vertx-web`

I took them from the "Installed features" log at the start, is it ok? 

The implementation of the Client is the class `ClientRequestContextImpl` from `resteasy-client.4.5.9-Final`.

I saw your PR but didn t thought was for rest clients too tbh
